lib/web_ui/lib/src/engine/bitmap_canvas.dart

```diff
@@ -524,6 +537,7 @@ class BitmapCanvas extends EngineCanvas {
       rootElement.append(paragraphElement);
     }
     _children.add(paragraphElement);
+    _allocateNewCanvas();
```

Comment:
We used to do this before, but then discovered that this killed performance on some of the early demos, such as the weather app demo, so we changed it to always put text on top. Can you please add a benchmark prior to submitting this PR that interleaves text with drawing commands so we understand the performance hit of this change? I imagine once @mdebbar's text on canvas work is done we'll be able to use `fillText` for all text.

Reply:
A while ago introduced childOverDraw which makes sure that once we switch to using <p> tags we don't go back requesting extra canvas(s) to draw text. The weather app had drawImage+drawText pattern which is already handled since images are drawn using <img> tags all the time now. The only benchmark we could artificially create that would show regression would be of the pattern  drawText, drawComplexPath, drawText that i have not seen in apps yet. One option : A list of material cards that have draw path + images + text is a more realistic benchmark but would not be effected by this change.

Reply:
Ah, right, we're creating them lazily. Ok, I guess it's not critical then, just nice-to-have.